src/exporter.js

```diff
@@ -5,38 +5,30 @@ const log = debug('exporter')
 log.err = debug('exporter:error')
 const UnixFS = require('ipfs-unixfs')
 const async = require('async')
-const events = require('events')
 const Readable = require('readable-stream').Readable
 const pathj = require('path')
+const util = require('util')
 
-exports = module.exports = exporter
+exports = module.exports = Exporter
 
-function exporter (hash, dagService, options, callback) {
-  if (typeof options === 'function') {
-    callback = options
-    options = {}
+util.inherits(Exporter, Readable)
+
+function Exporter (hash, dagService, options) {
+  Readable.call(this, { objectMode: true })
+
+  if (!(this instanceof Exporter)) {
```

Comment:
You might be leaking memory in a weird way if this check doesn't come before the `Readable.call(...)` line.